include/natalie/value.hpp

```diff
@@ -80,7 +86,21 @@ class Value {
     }
 
     bool operator!=(Value other) const {
-        return !(*this == other);
+        if (is_fast_integer()) {
+            if (other.is_fast_integer())
+                return get_fast_integer() != other.get_fast_integer();
+            if (other.is_fast_float())
+                return get_fast_integer() != other.get_fast_float();
+            return false;
```

Comment:
Neither this nor the other non-hydrating return false at line 101 is true, what if you are comparing with nulltpr? Or what if you are comparing with an hydrated int/float? I'd definitely suggest adding an explicit check for nullptr but also removing return false here (and the other way around for ==) and letting it fall-through a check with the hydrated objects at the end (or just simply fall-through the way it was before). Having said that I think it's a really good idea checking fast-non-hydrated-values :^)